### ISSUE TITLE
[2.0] fetch module: fail_on_missing is ignored

### ISSUE THREAD

**Issue by rvdh** [ansible/ansible#13832]
##### Ansible Version:

ansible 2.0.0.1
  config file = /Users/rick/.ansible.cfg
  configured module search path = Default w/o overrides
##### Ansible Configuration:

Nothing has changed other than that I've upgraded ansible from 1.9.4 to 2.0.0.1.
##### Environment:

This is on MacOS X 10.11.2 (El Capitan). 
##### Summary:

When running a playbook, which works fine on 1.9.4, on ansible 2.0.0.1 an error is generated by the fetch module:
(I've edited the 'item' bits because these contained customer information)

```
failed: [foo] => (item='foo') => {"failed": true, "item": "foo"}, "msg": "file not found: /root/.mysql/foo.txt"}
```
##### Steps To Reproduce:

Run the following task on ansible 2.0.0.1:

```
- name: download the credentials
  fetch:
    dest: credentials/
    src: "/root/.mysql/{{ item.name }}.txt"
    flat: true
    fail_on_missing: false
  with_items: "{{ users }}"
```
##### Expected Results:

The task should not fail. It does not fail on ansible 1.9.4:

```
TASK: [download the credentials] **********************************************
ok: [foo] => (item='foo')
```
##### Actual Results:

The task does fail on ansible 2.0.0.1:

```
TASK [download the credentials] ************************************************
failed: [foo] => (item='foo') => {"failed": true, "item": 'foo', "msg": "file not found: /root/.mysql/foo.txt"}
```

**Comment by abadger** [comment]
This is the minimal reproducer I could create:

```
---
- hosts: rhel7-test
  vars:
    users:
      - foo
  tasks:
  - fetch:
      dest: /var/tmp/test
      src: "/var/tmp/doesnotexist"
      flat: true
      fail_on_missing: false
    become: True
```

Without become, fail_on_missing works.

**Comment by abadger** [comment]
The above commit should fix this problem.  It's merged to devel and the stable-2.0 branch so it should be in the next 2.0.x release.

**Comment by rvdh** [comment]
Thanks @abadger!
